src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrame.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This class holds ALL types of delegates that are coverted from ONE target method
+    /// Type parameters are generated for: <see cref="GetTypeArgumentsFromConversion(TypeSymbol, MethodSymbol)"/>
+    ///     1. All type parameters from the containing types of the target method
+    ///     2. All type parameters from the target method
+    ///     3. The delegate type
+    /// </summary>
+    /// <remarks>
+    /// This symbol is created in lowering phrase, and is collected and translated in emit phrase.
+    /// </remarks>
+    internal sealed class MethodGroupConversionCacheFrame : NamedTypeSymbol
```

Comment:
Not convinced we need a new symbol for this. Perhaps we should simply keep track of method groups in lowering via a dictionary or other grouping mechanism and build a conversion cache that way.

I'm very skeptical a cache frame should be classified as a `NamedTypeSymbol`.